### PR TITLE
membership repo

### LINKED ISSUE #175: MySQL Membership Repository
refer to https://github.com/vinyldns/vinyldns/issues/16

1. Create MySQLMembershipRepository that implements functionality of DynamoDBMembershipRepository
1. Create unit and integration tests

### PR DESCRIPTION
**Pull request by nimaeskandary**
Fixes #175 .

Changes in this pull request:
- add mysql membership repo

### CODE REVIEW

File: modules/mysql/src/main/resources/db/migration/V3.6__Membership.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+USE ${dbName};
+
+/*
+Create table to store membership relationships
+*/
+CREATE TABLE membership (
+  user_id CHAR(36) NOT NULL,
+  group_id CHAR(36) NOT NULL,
+  PRIMARY KEY (user_id, group_id)
```
[the remark this answers is not in the document]
Comment (pauljamescleary) [reply modules/mysql/src/main/resources/db/migration/V3.6__Membership.sql]: in general avoid foreign keys as they hurt performance and make database evolutions and more "SQL / MYSQL" features very difficult.

They are used in zones as zone+zone_access is really viewed as a single table.  That said, even there we could have done without them.

File: modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import vinyldns.core.domain.membership.MembershipRepository
+import vinyldns.mysql.TestMySqlInstance
+import scalikejdbc._
+
+class MySqlMembershipRepositoryIntegrationSpec
+  extends WordSpec
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach
+    with Matchers
+    with Inspectors
```
Comment (rebstar6) [inline modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala]: are you using Inspectors or OptionValues?

File: modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+      addResult should contain theSameElementsAs userIds
+
+      val getAllResult = getAllRecords
+      val expectedGetAllResult = Set(Tuple2(userIds.head, groupId))
+
+      getAllResult should contain theSameElementsAs expectedGetAllResult
+    }
+
+    "add multiple members successfully" in {
```
Comment (rebstar6) [inline modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala]: test that duplicates are fine. so add (grp1, user1) twice and validate that it's all good

File: modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala (hunk 2 above)
Comment (nimaeskandary) [reply modules/mysql/src/it/scala/vinyldns/mysql/repository/MySqlMembershipRepositoryIntegrationSpec.scala]: nope, will remove

**Pull request merged**